runner/runner.go

```diff
@@ -1329,6 +1330,48 @@ func Grade(
 						},
 					)
 				}
+				// Check if we have an expected stderr message from the validator;
+				// fail the case with JE if there's a mismatch.
+				if runScore.Cmp(big.NewRat(1, 1)) != 0 {
+					expectedStderrPath := path.Join(
+						input.Path(), "cases", fmt.Sprintf("%s.expected-failure", caseData.Name),
+					)
+					expectedStderr, err := ioutil.ReadFile(expectedStderrPath)
+					if err != nil && !errors.Is(err, fs.ErrNotExist) {
+						ctx.Log.Warn(
+							"Error opening expected file",
+							map[string]interface{}{
+								"path": expectedStderrPath,
+								"err":  err,
+							},
+						)
+						continue
+					}
+					if err == nil {
```

Comment:
se puede cambiar para que se use el [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)?

```suggestion
					if err != nil {
						// Log
						continue
					}
					// Lo demás.
```

también se puede colapsar con el bloque de arriba para sólo logguear cuando no es un error de no-encontrado, pero siempre continuar si el error no es nulo.